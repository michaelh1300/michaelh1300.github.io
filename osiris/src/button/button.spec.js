import { mount } from '@vue/test-utils';
import Button from './button.vue';

describe('Button', () => {
  let wrapper;

  const reset = () => wrapper.setProps({
    type: 'default',
    nativeType: 'button',
    size: 'normal',
    round: false,
    circle: false,
    disabled: false,
  });

  beforeAll(() => {
    wrapper = mount(Button);
  });

  afterEach(reset);

  it('should be a default button at startup', () => {
    const defaultClasses = ['o-button', 'o-button--default', 'o-button--normal'];

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toEqual(defaultClasses);
  });

  it('should be a primary button', () => {
    wrapper.setProps({
      type: 'primary',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-button--primary');
  });

  it('should be a secondary button', () => {
    wrapper.setProps({
      type: 'secondary',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-button--secondary');
  });

  it('should be a info button', () => {
    wrapper.setProps({
      type: 'info',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-button--info');
  });

  it('should be a success button', () => {
    wrapper.setProps({
      type: 'success',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-button--success');
  });

  it('should be a warning button', () => {
    wrapper.setProps({
      type: 'warning',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-button--warning');
  });

  it('should be a danger button', () => {
    wrapper.setProps({
      type: 'danger',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-button--danger');
  });

  it('should be a text button', () => {
    wrapper.setProps({
      type: 'text',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-button--text');
  });

  it('should be a mini button', () => {
    wrapper.setProps({
      size: 'mini',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-button--mini');
  });

  it('should be a small button', () => {
    wrapper.setProps({
      size: 'small',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-button--small');
  });

  it('should be a normal button', () => {
    wrapper.setProps({
      size: 'normal',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-button--normal');
  });

  it('should be a large button', () => {
    wrapper.setProps({
      size: 'large',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-button--large');
  });

  it('should be a round button', () => {
    wrapper.setProps({
      round: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('is-round');
  });

  it('should be a circle button', () => {
    wrapper.setProps({
      circle: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('is-circle');
  });

  it('should be a plain button', () => {
    wrapper.setProps({
      plain: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('is-plain');
  });

  it('should have a nativeType submit', () => {
    wrapper.setProps({
      nativeType: 'submit',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.attributes().type).toBe('submit');
  });

  it('should be disabled', () => {
    wrapper.setProps({
      disabled: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.attributes().disabled).toBeTruthy();
  });

  it('should emit click event', () => {
    wrapper.findAll('button').trigger('click');
    expect(wrapper.emitted().click).toBeTruthy();
  });

  it('should render default slot', () => {
    wrapper = mount(Button, {
      slots: {
        default: 'Default Button',
      },
    });

    expect(wrapper.text()).toBe('Default Button');
  });
});
