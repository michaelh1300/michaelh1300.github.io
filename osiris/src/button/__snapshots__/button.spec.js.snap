// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be a circle button 1`] = `<button type="button" class="o-button o-button--default o-button--normal is-circle"></button>`;

exports[`Button should be a danger button 1`] = `<button type="button" class="o-button o-button--danger o-button--normal"></button>`;

exports[`Button should be a default button at startup 1`] = `<button type="button" class="o-button o-button--default o-button--normal"></button>`;

exports[`Button should be a info button 1`] = `<button type="button" class="o-button o-button--info o-button--normal"></button>`;

exports[`Button should be a large button 1`] = `<button type="button" class="o-button o-button--default o-button--large"></button>`;

exports[`Button should be a mini button 1`] = `<button type="button" class="o-button o-button--default o-button--mini"></button>`;

exports[`Button should be a normal button 1`] = `<button type="button" class="o-button o-button--default o-button--normal"></button>`;

exports[`Button should be a plain button 1`] = `<button type="button" class="o-button o-button--default o-button--normal is-plain"></button>`;

exports[`Button should be a primary button 1`] = `<button type="button" class="o-button o-button--primary o-button--normal"></button>`;

exports[`Button should be a round button 1`] = `<button type="button" class="o-button o-button--default o-button--normal is-round"></button>`;

exports[`Button should be a secondary button 1`] = `<button type="button" class="o-button o-button--secondary o-button--normal"></button>`;

exports[`Button should be a small button 1`] = `<button type="button" class="o-button o-button--default o-button--small"></button>`;

exports[`Button should be a success button 1`] = `<button type="button" class="o-button o-button--success o-button--normal"></button>`;

exports[`Button should be a text button 1`] = `<button type="button" class="o-button o-button--text o-button--normal"></button>`;

exports[`Button should be a warning button 1`] = `<button type="button" class="o-button o-button--warning o-button--normal"></button>`;

exports[`Button should be disabled 1`] = `<button type="button" class="o-button o-button--default o-button--normal is-disabled is-plain" disabled="disabled"></button>`;

exports[`Button should have a nativeType submit 1`] = `<button type="submit" class="o-button o-button--default o-button--normal is-plain"></button>`;
